app/models/miq_worker/runner.rb

```diff
@@ -177,7 +177,6 @@ def find_worker_record
 
   def starting_worker_record
     find_worker_record
-    @worker.pid            = Process.pid
```

Comment:
This is important, as the worker model will store the `self.pid=` with the result of MiqWorker#start_runner, which when changed to return the Thread identifier(an object_id), gets overwritten by this line 180. 

We need to remove most places making assumptions about calling Process.pid for the workers.